components/notebook-controller/Dockerfile

```diff
@@ -3,32 +3,60 @@
 # The Docker context is expected to be:
 #
 # ${PATH_TO_KUBEFLOW/KUBEFLOW repo}/components
-#
-# This is necessary because the Jupyter controller now depends on
-# components/common
+
+
+# Build arguments
+ARG SOURCE_CODE=.
 ARG GOLANG_VERSION=1.20
-FROM golang:${GOLANG_VERSION} as builder
 
+# Use ubi8/go-toolset as base image
+FROM registry.redhat.io/ubi8/go-toolset:${GOLANG_VERSION} as builder
+
+## Build args to be used at this step
+ARG SOURCE_CODE
+
+# Set building workdir
 WORKDIR /workspace
 
 # Copy the Go Modules manifests
-COPY notebook-controller /workspace/notebook-controller
-COPY common /workspace/common
-
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN cd /workspace/notebook-controller && go mod download all
+COPY ${SOURCE_CODE}/notebook-controller ./notebook-controller
+# This is necessary because the Jupyter controller now depends on
+# components/common
+COPY ${SOURCE_CODE}/common ./common
 
+# Update building workdir
 WORKDIR /workspace/notebook-controller
 
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -mod=mod -o manager main.go
+## Build the kf-notebook-controller
+USER root
+
+RUN if [ -z ${CACHITO_ENV_FILE} ]; then \
+       go mod download all; \
+    else \
+       source ${CACHITO_ENV_FILE}; \
+    fi
+
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -mod=mod \
+        -o ./bin/manager main.go
+
+# Use ubi8/ubi-minimal as base image
+FROM registry.redhat.io/ubi8/ubi-minimal:latest
+
+## Install additional packages
+RUN microdnf install -y shadow-utils &&\
+    microdnf clean all
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/base:debug
-WORKDIR /
-COPY --from=builder /workspace/notebook-controller/manager .
+## Create a non-root user with UID 1001
+RUN useradd --uid 1001 --create-home --user-group --system rhods
+
+## Set workdir directory to user home
+WORKDIR /home/rhods
```

Comment:
Do we like this branding in an upstream?

Reply:
I mentioned this myself in https://github.com/opendatahub-io/kubeflow/pull/340#discussion_r1627423852. @VaishnaviHire mentioned opendatahub-io is midstream, so I think we can afford this.

Reply:
It's like we depend on the oauth proxy image that is only available with redhat subscription, not on unregistered okd, and odh needs operators in the redhat catalog that aren't in the community catalog

Reply:
Yeah... this is all insane... I wonder who installs ODH since they need to have access to OpenShift and Red Hat catalogue anyway... why not to use properly supported RHOAI... only reason I could possibly see are the new features there that haven't been ported to RHOAI yet... which shall change...

Anyway, I don't have strict opinion... but we use different naming in ODH and RHOAI in some places - e.g. dashboard pods are named as `odh-dahsboard` in ODH and `rhods-dashboard` in RHOAI. On the other hand here in IDE scrum, we keep the same name of notebook manager with `odh-` prefix both in RHOAI and ODH... this change just backports naming from RHOAI back to ODH... so we are doing a completely disgusting mix...

Reply:
Just saying... I also like things to be same for ease and convenience... it's just that it's a total mess and chaos here. Feel free to ignore this thread.

Reply:
Aren't we planning on abandoning the @red-hat-data-science repos in the future (cc @adelton). How are we going to be doing this kind of "branding" then? Changing the username in dockerfile seems kind of silly and pointless waste of effort to me, best to have some neutral name that can be used everywhere the same.